drivers/Makefile.dep

```diff
@@ -521,6 +521,7 @@ ifneq (,$(filter sx127%,$(USEMODULE)))
   USEMODULE += sx127x
   USEMODULE += netif
   USEMODULE += lora
+  USEMODULE += netif_cfg_lora
```

Comment:
I don't like the name:

1) The shell command is `gnrc_netif` specific, not `netif`.
2) `cfg` implies some configuration related stuff, while it is only related to `shell_commands`

How about `gnrc_netif_cmd_lora` and `gnrc_netif_cmd_%` for the generic version?

Reply:
I used `gnrc_netif_phy_%` before, however I noticed a problem: [`Makefile.dep`](https://github.com/RIOT-OS/RIOT/blob/master/Makefile.dep#L100) matches `gnrc_%` and then includes the `gnrc` module.

This leads to a build failure of `tests/driver_sx127x`.

So I had to look for another prefix.

Since this is only relevant for the `ifconfig` shell command, I choose the `shell_` prefix which is still 'save'.

Reply:
Yea, I should read on some more… I can just add it to the `filter-out` function there.

Reply:
> I used `gnrc_netif_phy_%` before, however I noticed a problem: [`Makefile.dep`](https://github.com/RIOT-OS/RIOT/blob/master/Makefile.dep#L100) matches `gnrc_%` and then includes the `gnrc` module.

This seems to me to be some weird dependency issue. What exactly is failing for the test in question?

Reply:
Ah, because you should only include `gnrc_netif_cmd_` (or what ever) when `gnrc` is present ;-). So the dependency issue was introduced in your own PR ;-).

Reply:
> Yea, I should read on some more… I can just add it to the `filter-out` function there.

(saw this comment only after I posted)

Reply:
```
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /home/benpicco/dev/RIOT/tests/driver_sx127x/bin/samr21-xpro/shell_commands/sc_gnrc_netif.o: in function `gnrc_netapi_get':
/home/benpicco/dev/RIOT/sys/include/net/gnrc/netapi.h:236: undefined reference to `_gnrc_netapi_get_set'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /home/benpicco/dev/RIOT/tests/driver_sx127x/bin/samr21-xpro/shell_commands/sc_gnrc_netif.o: in function `gnrc_netapi_set':
/home/benpicco/dev/RIOT/sys/include/net/gnrc/netapi.h:261: undefined reference to `_gnrc_netapi_get_set'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /home/benpicco/dev/RIOT/sys/include/net/gnrc/netapi.h:261: undefined reference to `_gnrc_netapi_get_set'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /home/benpicco/dev/RIOT/sys/include/net/gnrc/netapi.h:261: undefined reference to `_gnrc_netapi_get_set'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /home/benpicco/dev/RIOT/sys/include/net/gnrc/netapi.h:261: undefined reference to `_gnrc_netapi_get_set'
/usr/lib/gcc/arm-none-eabi/7.3.1/../../../arm-none-eabi/bin/ld: /home/benpicco/dev/RIOT/tests/driver_sx127x/bin/samr21-xpro/shell_commands/sc_gnrc_netif.o:/home/benpicco/dev/RIOT/sys/include/net/gnrc/netapi.h:261: more undefined references to `_gnrc_netapi_get_set' follow
```

Reply:
Yeah. Because your PR puts GNRC dependencies in when they are actually not needed. So:

```suggestion
  ifneq (,$(filter gnrc,$(USEMODULE))
    # Pull in `ifconfig` support for LoRA
    USEMODULE += gnrc_netif_cmd_lora
  endif
```

Reply:
good point!